src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentServiceInternalImpl.java

```
@@ -292,10 +300,80 @@ public List<ItemVersion> getContentVersionHistory(String siteId, String path) th
         }
     }
 
+    public boolean createFolder(String siteId, String path, String name) throws ServiceLayerException, UserNotFoundException {
+        String folderPath = path + FILE_SEPARATOR + name;
+        boolean toRet = false;
+        String commitId = contentRepository.createFolder(siteId, path, name);
+        if (commitId != null) {
```

ContentRepository.createFolder should never return null. Instead, it should throw an Exception if an error happens

---

I updated to throw exceptions instead of returning `null`